internal/testutil/datagen/README.md

```diff
@@ -0,0 +1,90 @@
+
+## Requirements
+
+* `kubectl` and `helm` installed and configured
+* Current `kubectl` context pointed to a preferably freshly-created minikube cluster.
+
+## How it works
+
+`datagen.sh` script will deploy a series of dummy and required workloads to the cluster, making it a cluster containing at least one resource of all the resources we can possibly monitor.
+
+More specifically, it will deploy:
+- The KSM chart, for the KSM services/endpoints discovery tests
+- The [`e2e-resources`](../../../e2e/charts/e2e-resources) chart, which includes multiple samples of resources we monitor:
+  - Dummy Pods, Services, StatefulSets, DaemonSets and Deployments in several states
+  - HPA targets
+  - If enabled, PersistentVolumes and PersistentVolumeClaims
+
+After this, it will deploy KSM and programmatically hit all the endpoints required for the integration to work, and store them in a directory specified by the user:
+
+- Kubelet endpoints
+  - `/pods` 
+  - `/stats/summary`
+  - `/metrics/cadvisor`
+- KSM `/metrics`
+- Control plane components
+  - api-server `/metrics`
+  - controller-manager `/metrics`
+  - etcd `/metrics`
+  - scheduler `/metrics`
+
+It will do this by spawning a privileged, `hostNetwork` `alpine:latest` pod in the cluster and running itself from inside.
+
+## Usage
+
+```
+./datagen.sh <output_folder>
+```
+
+Output folder would typically be a version number, e.g.
+
+```shell
+./datagen.sh 1_22
+```
+
+### Arguments and config
+
+#### Endpoints
+
+Endpoints for the targets to scrape can be overridden using environment variables:
+
+```shell
+KSM_ENDPOINT=${KSM_ENDPOINT:-http://ksm-kube-state-metrics.ksm.svc:8080/metrics}
+KUBELET_ENDPOINT=${KUBELET_ENDPOINT:-https://localhost:10250/}
+ETCD_ENDPOINT=${ETCD_ENDPOINT:-http://localhost:2381/metrics}
+APISERVER_ENDPOINT=${APISERVER_ENDPOINT:-https://localhost:8443/metrics}
+CONTROLLERMANAGER_ENDPOINT=${CONTROLLERMANAGER_ENDPOINT:-https://localhost:10257/metrics}
+SCHEDULER_ENDPOINT=${SCHEDULER_ENDPOINT:-https://localhost:10259/metrics}
+```
+
+#### `DISABLE_CONTROLPLANE`
+
+By default, `scraper.sh` will attempt to reach the controlpane endpoints through `localhost`. This will not work on managed K8s environments like EKS or GKE. By setting `DISABLE_CONTROLPLANE` to a non-empty value, `scraper.sh` will not attempt to reach control plane components.
+
+> Note: For scraping EKS/GKE clusters, it's also recommended to set `IS_MINIKUBE` to false (see below). 
+
+#### `IS_MINIKUBE`
+
+Aiming to be extensive by default, `datagen.sh` will assume it runs in a minikube environment. This means it will:
+- Try to run `minikube addons enable metrics-server`
+- Deploy some PVCs with storageClass set to standard (minikube default)
+- Deploy the `LoadBalancer` service with a fake `loadBalancerIP`
```

Comment:
Is this needed to collect some metrics?

Reply:
Correct! https://github.com/newrelic/nri-kubernetes/blob/5a318a81ae3275fbc99140fc206050e8b143a4fb/src/metric/definition.go#L602

One less exception in the integration test :)